transport/src/main/java/io/scalecube/transport/Addressing.java

```diff
@@ -0,0 +1,210 @@
+package io.scalecube.transport;
+
+import com.google.common.base.Strings;
+import com.google.common.base.Throwables;
+
+import java.io.IOException;
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.Socket;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * Utility class which finds local IP address and currently available server ports.
+ *
+ * @author Anton Kharenko
+ * @author Artem Vysochyn
+ */
+class Addressing {
+
+  /**
+   * The minimum server port number. Set at 1100 to avoid returning privileged port numbers.
+   */
+  static final int MIN_PORT_NUMBER = 1100;
+
+  /**
+   * The maximum server port number. It should only use up to 49151, as from 49152 up to 65535 are reserved for
+   * ephemeral ports.
+   */
+  static final int MAX_PORT_NUMBER = 49151;
+
+  private Addressing() {}
+
+  /**
+   * Finds next available port starting from given port and incrementing port for port count iterations.
+   *
+   * @param fromPort starting port
+   * @param portCount number of ports to check
+   * @return Next available port
+   * @throws NoSuchElementException if port wasn't found in the given interval
+   */
+  static int getNextAvailable(int fromPort, int portCount) {
+    int toPort = Math.min(MAX_PORT_NUMBER, fromPort + portCount) - 1;
+    for (int port = fromPort; port <= toPort; port++) {
+      if (isAvailable(port)) {
+        return port;
+      }
+    }
+    throw new NoSuchElementException("Could not find an available port from " + fromPort + " to " + toPort);
+  }
+
+  /**
+   * Checks to see if a specific port is available.
+   *
+   * @param port the port to check for availability
+   */
+  static boolean isAvailable(int port) {
+    if (port < MIN_PORT_NUMBER || port > MAX_PORT_NUMBER) {
+      throw new IllegalArgumentException("Invalid port number: " + port);
+    }
+
+    Socket socket = null;
+    try {
+      socket = new Socket(getLocalIpAddress(), port);
```

Comment:
hm, once we start to resolve IP address this logic is not really correct anymore. First we need to resolve listen address we are going to bind and then check available port for this specific resolved address. So in other words pass address here and to getNextAvailable{Port}() as one of arguments.